[{
	"resource": "/e:/kp/python/InvoiceApp/main.py",
	"owner": "python",
	"code": {
		"value": "reportIncompatibleMethodOverride",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Method \"showMessage\" overrides class \"QSplashScreen\" in an incompatible manner\n  Parameter 2 type mismatch: base parameter is type \"str | None\", override parameter is type \"str\"\n    Type \"str | None\" is not assignable to type \"str\"\n      \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 150,
	"startColumn": 9,
	"endLineNumber": 150,
	"endColumn": 20,
	"relatedInformation": [
		{
			"startLineNumber": 6588,
			"startColumn": 9,
			"endLineNumber": 6588,
			"endColumn": 20,
			"message": "Overridden method",
			"resource": "/e:/kp/python/InvoiceApp/venv/Lib/site-packages/PyQt6/QtWidgets.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 26,
	"startColumn": 50,
	"endLineNumber": 26,
	"endColumn": 54
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"Dict[Unknown, Unknown]\"\n  \"None\" is not assignable to \"Dict[Unknown, Unknown]\"",
	"source": "Pylance",
	"startLineNumber": 26,
	"startColumn": 72,
	"endLineNumber": 26,
	"endColumn": 76
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"int\" cannot be assigned to parameter \"subtotal\" of type \"Decimal\" in function \"calculate_vat_amount\"\n  \"int\" is not assignable to \"Decimal\"",
	"source": "Pylance",
	"startLineNumber": 60,
	"startColumn": 48,
	"endLineNumber": 60,
	"endColumn": 56
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal]\" cannot be assigned to parameter \"vat_percentage\" of type \"Decimal\" in function \"calculate_vat_amount\"\n  \"Column[Decimal]\" is not assignable to \"Decimal\"",
	"source": "Pylance",
	"startLineNumber": 60,
	"startColumn": 58,
	"endLineNumber": 60,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"dict[str, int | Decimal]\" is not assignable to return type \"Dict[str, Decimal]\"\n  \"int\" is not assignable to \"Decimal\"",
	"source": "Pylance",
	"startLineNumber": 64,
	"startColumn": 25,
	"endLineNumber": 64,
	"endColumn": 33
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"subtotal\" for class \"Invoice\"\n  Expression of type \"Decimal\" cannot be assigned to attribute \"subtotal\" of class \"Invoice\"\n    Type \"Decimal\" is not assignable to type \"Column[Decimal] | int\"\n      \"Decimal\" is not assignable to \"Column[Decimal]\"\n      \"Decimal\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 72,
	"startColumn": 17,
	"endLineNumber": 72,
	"endColumn": 25
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 76,
	"startColumn": 60,
	"endLineNumber": 76,
	"endColumn": 64
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"current_number\" for class \"InvoiceNumberSequence\"\n  Expression of type \"ColumnElement[int]\" cannot be assigned to attribute \"current_number\" of class \"InvoiceNumberSequence\"\n    \"ColumnElement[int]\" is not assignable to \"Column[int]\"",
	"source": "Pylance",
	"startLineNumber": 102,
	"startColumn": 18,
	"endLineNumber": 102,
	"endColumn": 32
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"sequence\" of type \"int\" in function \"format_invoice_number\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 105,
	"startColumn": 61,
	"endLineNumber": 105,
	"endColumn": 84
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"current_number\" for class \"InvoiceNumberSequence\"\n  Expression of type \"ColumnElement[int]\" cannot be assigned to attribute \"current_number\" of class \"InvoiceNumberSequence\"\n    \"ColumnElement[int]\" is not assignable to \"Column[int]\"",
	"source": "Pylance",
	"startLineNumber": 114,
	"startColumn": 22,
	"endLineNumber": 114,
	"endColumn": 36
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"sequence\" of type \"int\" in function \"format_invoice_number\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 115,
	"startColumn": 65,
	"endLineNumber": 115,
	"endColumn": 88
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"BinaryExpression[bool] | Literal[True]\" cannot be assigned to parameter \"clauses\" of type \"_ColumnExpressionArgument[bool]\" in function \"and_\"\n  Type \"BinaryExpression[bool] | Literal[True]\" is not assignable to type \"_ColumnExpressionArgument[bool]\"\n    Type \"Literal[True]\" is not assignable to type \"_ColumnExpressionArgument[bool]\"\n      \"Literal[True]\" is not assignable to \"ColumnElement[bool]\"\n      \"Literal[True]\" is incompatible with protocol \"_HasClauseElement[bool]\"\n        \"__clause_element__\" is not present\n      \"Literal[True]\" is not assignable to \"SQLCoreOperations[bool]\"\n      \"Literal[True]\" is not assignable to \"ExpressionElementRole[bool]\"\n      \"Literal[True]\" is not assignable to \"TypedColumnsClauseRole[bool]\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 196,
	"startColumn": 21,
	"endLineNumber": 196,
	"endColumn": 70
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"text\" of type \"str\" in function \"fuzzy_search_score\"\n  \"Column[str]\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 204,
	"startColumn": 47,
	"endLineNumber": 204,
	"endColumn": 67
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"text\" of type \"str\" in function \"fuzzy_search_score\"\n  \"Column[str]\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 205,
	"startColumn": 47,
	"endLineNumber": 205,
	"endColumn": 59
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"text\" of type \"str\" in function \"fuzzy_search_score\"\n  \"Column[str]\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 206,
	"startColumn": 47,
	"endLineNumber": 206,
	"endColumn": 60
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "No overloads for \"__init__\" match the provided arguments",
	"source": "Pylance",
	"startLineNumber": 338,
	"startColumn": 30,
	"endLineNumber": 338,
	"endColumn": 49,
	"relatedInformation": [
		{
			"startLineNumber": 1116,
			"startColumn": 9,
			"endLineNumber": 1116,
			"endColumn": 17,
			"message": "Overload 8 is the closest match",
			"resource": "/c:/Users/A/.vscode/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"List[Row[Tuple[str, int]]]\" cannot be assigned to parameter \"iterable\" of type \"Iterable[list[bytes]]\" in function \"__init__\"\n  \"List[Row[Tuple[str, int]]]\" is not assignable to \"Iterable[list[bytes]]\"\n    Type parameter \"_T_co@Iterable\" is covariant, but \"Row[Tuple[str, int]]\" is not a subtype of \"list[bytes]\"\n      \"Row[Tuple[str, int]]\" is not assignable to \"list[bytes]\"",
	"source": "Pylance",
	"startLineNumber": 338,
	"startColumn": 35,
	"endLineNumber": 338,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "No overloads for \"__init__\" match the provided arguments",
	"source": "Pylance",
	"startLineNumber": 339,
	"startColumn": 31,
	"endLineNumber": 339,
	"endColumn": 51,
	"relatedInformation": [
		{
			"startLineNumber": 1116,
			"startColumn": 9,
			"endLineNumber": 1116,
			"endColumn": 17,
			"message": "Overload 8 is the closest match",
			"resource": "/c:/Users/A/.vscode/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"List[Row[Tuple[str, Column[Decimal] | int | Unknown]]]\" cannot be assigned to parameter \"iterable\" of type \"Iterable[list[bytes]]\" in function \"__init__\"\n  \"List[Row[Tuple[str, Column[Decimal] | int | Unknown]]]\" is not assignable to \"Iterable[list[bytes]]\"\n    Type parameter \"_T_co@Iterable\" is covariant, but \"Row[Tuple[str, Column[Decimal] | int | Unknown]]\" is not a subtype of \"list[bytes]\"\n      \"Row[Tuple[str, Column[Decimal] | int | Unknown]]\" is not assignable to \"list[bytes]\"",
	"source": "Pylance",
	"startLineNumber": 339,
	"startColumn": 36,
	"endLineNumber": 339,
	"endColumn": 50
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "No overloads for \"__init__\" match the provided arguments",
	"source": "Pylance",
	"startLineNumber": 340,
	"startColumn": 31,
	"endLineNumber": 340,
	"endColumn": 51,
	"relatedInformation": [
		{
			"startLineNumber": 1116,
			"startColumn": 9,
			"endLineNumber": 1116,
			"endColumn": 17,
			"message": "Overload 8 is the closest match",
			"resource": "/c:/Users/A/.vscode/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"List[Row[Tuple[int, Column[Decimal] | int | Unknown]]]\" cannot be assigned to parameter \"iterable\" of type \"Iterable[list[bytes]]\" in function \"__init__\"\n  \"List[Row[Tuple[int, Column[Decimal] | int | Unknown]]]\" is not assignable to \"Iterable[list[bytes]]\"\n    Type parameter \"_T_co@Iterable\" is covariant, but \"Row[Tuple[int, Column[Decimal] | int | Unknown]]\" is not a subtype of \"list[bytes]\"\n      \"Row[Tuple[int, Column[Decimal] | int | Unknown]]\" is not assignable to \"list[bytes]\"",
	"source": "Pylance",
	"startLineNumber": 340,
	"startColumn": 36,
	"endLineNumber": 340,
	"endColumn": 50
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 343,
	"startColumn": 73,
	"endLineNumber": 343,
	"endColumn": 77
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 363,
	"startColumn": 65,
	"endLineNumber": 363,
	"endColumn": 69
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 374,
	"startColumn": 67,
	"endLineNumber": 374,
	"endColumn": 71
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 392,
	"startColumn": 73,
	"endLineNumber": 392,
	"endColumn": 77
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 405,
	"startColumn": 65,
	"endLineNumber": 405,
	"endColumn": 69
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 418,
	"startColumn": 67,
	"endLineNumber": 418,
	"endColumn": 71
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 485,
	"startColumn": 64,
	"endLineNumber": 485,
	"endColumn": 68
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"setting_value\" for class \"Setting\"\n  Expression of type \"str\" cannot be assigned to attribute \"setting_value\" of class \"Setting\"\n    \"str\" is not assignable to \"Column[str]\"",
	"source": "Pylance",
	"startLineNumber": 492,
	"startColumn": 21,
	"endLineNumber": 492,
	"endColumn": 34
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"updated_by\" for class \"Setting\"\n  Expression of type \"int\" cannot be assigned to attribute \"updated_by\" of class \"Setting\"\n    \"int\" is not assignable to \"Column[int]\"",
	"source": "Pylance",
	"startLineNumber": 493,
	"startColumn": 21,
	"endLineNumber": 493,
	"endColumn": 31
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 523,
	"startColumn": 58,
	"endLineNumber": 523,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 523,
	"startColumn": 81,
	"endLineNumber": 523,
	"endColumn": 85
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"List[int]\"\n  \"None\" is not assignable to \"List[int]\"",
	"source": "Pylance",
	"startLineNumber": 524,
	"startColumn": 56,
	"endLineNumber": 524,
	"endColumn": 60
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 524,
	"startColumn": 76,
	"endLineNumber": 524,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 545,
	"startColumn": 64,
	"endLineNumber": 545,
	"endColumn": 68
},{
	"resource": "/e:/kp/python/InvoiceApp/models/business.py",
	"owner": "python",
	"code": {
		"value": "reportPossiblyUnboundVariable",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportPossiblyUnboundVariable.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"session\" is possibly unbound",
	"source": "Pylance",
	"startLineNumber": 607,
	"startColumn": 13,
	"endLineNumber": 607,
	"endColumn": 20
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"__tablename__\" for class \"type[UUIDMixin]*\"\n  Attribute \"__tablename__\" is unknown",
	"source": "Pylance",
	"startLineNumber": 39,
	"startColumn": 35,
	"endLineNumber": 39,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Object of type \"Column[str]\" is not callable\n  Attribute \"__call__\" is unknown",
	"source": "Pylance",
	"startLineNumber": 153,
	"startColumn": 18,
	"endLineNumber": 153,
	"endColumn": 76
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n    \"Column[Decimal]\" is not assignable to \"str\"\n    \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n      \"__float__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 206,
	"startColumn": 28,
	"endLineNumber": 206,
	"endColumn": 38
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"ColumnElement[Decimal]\" is not assignable to type \"ConvertibleToInt\"\n    \"ColumnElement[Decimal]\" is not assignable to \"str\"\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present\n  ...",
	"source": "Pylance",
	"startLineNumber": 283,
	"startColumn": 38,
	"endLineNumber": 283,
	"endColumn": 45
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"ColumnElement[Decimal]\" is not assignable to type \"ConvertibleToInt\"\n    \"ColumnElement[Decimal]\" is not assignable to \"str\"\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 283,
	"startColumn": 38,
	"endLineNumber": 283,
	"endColumn": 45
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"x\" of type \"SupportsAbs[_T@abs]\" in function \"abs\"\n  \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsAbs[_T@abs]\"\n    \"__abs__\" is not present",
	"source": "Pylance",
	"startLineNumber": 285,
	"startColumn": 16,
	"endLineNumber": 285,
	"endColumn": 35
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"ColumnElement[Decimal]\" is not assignable to type \"ConvertibleToInt\"\n    \"ColumnElement[Decimal]\" is not assignable to \"str\"\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present\n  ...",
	"source": "Pylance",
	"startLineNumber": 286,
	"startColumn": 35,
	"endLineNumber": 286,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"ColumnElement[Decimal]\" is not assignable to type \"ConvertibleToInt\"\n    \"ColumnElement[Decimal]\" is not assignable to \"str\"\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 286,
	"startColumn": 35,
	"endLineNumber": 286,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Invalid conditional operand of type \"ColumnElement[bool]\"\n  Method __bool__ for type \"ColumnElement[bool]\" returns type \"NoReturn\" rather than \"bool\"",
	"source": "Pylance",
	"startLineNumber": 287,
	"startColumn": 14,
	"endLineNumber": 287,
	"endColumn": 34
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"ColumnElement[Decimal]\" is not assignable to type \"ConvertibleToInt\"\n    \"ColumnElement[Decimal]\" is not assignable to \"str\"\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present\n  ...",
	"source": "Pylance",
	"startLineNumber": 288,
	"startColumn": 35,
	"endLineNumber": 288,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"ColumnElement[Decimal]\" is not assignable to type \"ConvertibleToInt\"\n    \"ColumnElement[Decimal]\" is not assignable to \"str\"\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"ColumnElement[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 288,
	"startColumn": 35,
	"endLineNumber": 288,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "No overloads for \"round\" match the provided arguments",
	"source": "Pylance",
	"startLineNumber": 290,
	"startColumn": 31,
	"endLineNumber": 290,
	"endColumn": 48,
	"relatedInformation": [
		{
			"startLineNumber": 1763,
			"startColumn": 5,
			"endLineNumber": 1763,
			"endColumn": 10,
			"message": "Overload 2 is the closest match",
			"resource": "/c:/Users/A/.vscode/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"ColumnElement[Decimal]\" cannot be assigned to parameter \"number\" of type \"_SupportsRound2[_T@round]\" in function \"round\"\n  \"ColumnElement[Decimal]\" is incompatible with protocol \"_SupportsRound2[_T@round]\"\n    \"__round__\" is not present",
	"source": "Pylance",
	"startLineNumber": 290,
	"startColumn": 37,
	"endLineNumber": 290,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Invalid conditional operand of type \"Column[date]\"\n  Method __bool__ for type \"Column[date]\" returns type \"NoReturn\" rather than \"bool\"",
	"source": "Pylance",
	"startLineNumber": 299,
	"startColumn": 62,
	"endLineNumber": 299,
	"endColumn": 79
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 300,
	"startColumn": 31,
	"endLineNumber": 300,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"",
	"source": "Pylance",
	"startLineNumber": 300,
	"startColumn": 31,
	"endLineNumber": 300,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n    \"Column[Decimal]\" is not assignable to \"str\"\n    \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n      \"__float__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 301,
	"startColumn": 37,
	"endLineNumber": 301,
	"endColumn": 56
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 302,
	"startColumn": 33,
	"endLineNumber": 302,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"",
	"source": "Pylance",
	"startLineNumber": 302,
	"startColumn": 33,
	"endLineNumber": 302,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 303,
	"startColumn": 35,
	"endLineNumber": 303,
	"endColumn": 52
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"",
	"source": "Pylance",
	"startLineNumber": 303,
	"startColumn": 35,
	"endLineNumber": 303,
	"endColumn": 52
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n    \"Column[Decimal]\" is not assignable to \"str\"\n    \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n      \"__float__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 357,
	"startColumn": 35,
	"endLineNumber": 357,
	"endColumn": 52
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Invalid conditional operand of type \"Column[Decimal]\"\n  Method __bool__ for type \"Column[Decimal]\" returns type \"NoReturn\" rather than \"bool\"",
	"source": "Pylance",
	"startLineNumber": 357,
	"startColumn": 57,
	"endLineNumber": 357,
	"endColumn": 74
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n    \"Column[Decimal]\" is not assignable to \"str\"\n    \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n      \"__float__\" is not present\n    \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 359,
	"startColumn": 33,
	"endLineNumber": 359,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | ColumnElement[int]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | ColumnElement[int]\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 360,
	"startColumn": 33,
	"endLineNumber": 360,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | ColumnElement[int]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | ColumnElement[int]\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"",
	"source": "Pylance",
	"startLineNumber": 360,
	"startColumn": 33,
	"endLineNumber": 360,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Invalid conditional operand of type \"ColumnElement[bool]\"\n  Method __bool__ for type \"ColumnElement[bool]\" returns type \"NoReturn\" rather than \"bool\"",
	"source": "Pylance",
	"startLineNumber": 385,
	"startColumn": 12,
	"endLineNumber": 385,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"Column[str]\" is not assignable to type \"ConvertibleToInt\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"Column[str]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"Column[str]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present\n  ...",
	"source": "Pylance",
	"startLineNumber": 386,
	"startColumn": 24,
	"endLineNumber": 386,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"x\" of type \"ConvertibleToInt\" in function \"__new__\"\n  Type \"Column[str]\" is not assignable to type \"ConvertibleToInt\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is incompatible with protocol \"Buffer\"\n      \"__buffer__\" is not present\n    \"Column[str]\" is incompatible with protocol \"SupportsInt\"\n      \"__int__\" is not present\n    \"Column[str]\" is incompatible with protocol \"SupportsIndex\"\n      \"__index__\" is not present",
	"source": "Pylance",
	"startLineNumber": 386,
	"startColumn": 24,
	"endLineNumber": 386,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Invalid conditional operand of type \"ColumnElement[bool]\"\n  Method __bool__ for type \"ColumnElement[bool]\" returns type \"NoReturn\" rather than \"bool\"",
	"source": "Pylance",
	"startLineNumber": 387,
	"startColumn": 14,
	"endLineNumber": 387,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"value\" of type \"_DecimalNew\" in function \"__new__\"\n  Type \"Column[str]\" is not assignable to type \"_DecimalNew\"\n    \"Column[str]\" is not assignable to \"Decimal\"\n    \"Column[str]\" is not assignable to \"float\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"tuple[int, Sequence[int], int]\"",
	"source": "Pylance",
	"startLineNumber": 388,
	"startColumn": 28,
	"endLineNumber": 388,
	"endColumn": 46
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Invalid conditional operand of type \"ColumnElement[bool]\"\n  Method __bool__ for type \"ColumnElement[bool]\" returns type \"NoReturn\" rather than \"bool\"",
	"source": "Pylance",
	"startLineNumber": 389,
	"startColumn": 14,
	"endLineNumber": 389,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalCall",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalCall.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Object of type \"None\" cannot be called",
	"source": "Pylance",
	"startLineNumber": 511,
	"startColumn": 20,
	"endLineNumber": 511,
	"endColumn": 39
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "No overloads for \"execute\" match the provided arguments",
	"source": "Pylance",
	"startLineNumber": 527,
	"startColumn": 13,
	"endLineNumber": 527,
	"endColumn": 40,
	"relatedInformation": [
		{
			"startLineNumber": 2294,
			"startColumn": 9,
			"endLineNumber": 2294,
			"endColumn": 16,
			"message": "Overload 3 is the closest match",
			"resource": "/e:/kp/python/InvoiceApp/venv/Lib/site-packages/sqlalchemy/orm/session.py"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/models/database.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Literal['SELECT 1']\" cannot be assigned to parameter \"statement\" of type \"Executable\" in function \"execute\"\n  \"Literal['SELECT 1']\" is not assignable to \"Executable\"",
	"source": "Pylance",
	"startLineNumber": 527,
	"startColumn": 29,
	"endLineNumber": 527,
	"endColumn": 39
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"Image\" is unknown import symbol",
	"source": "Pylance",
	"startLineNumber": 20,
	"startColumn": 30,
	"endLineNumber": 20,
	"endColumn": 35
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 128,
	"startColumn": 66,
	"endLineNumber": 128,
	"endColumn": 70
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 128,
	"startColumn": 89,
	"endLineNumber": 128,
	"endColumn": 93
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"List[int]\"\n  \"None\" is not assignable to \"List[int]\"",
	"source": "Pylance",
	"startLineNumber": 129,
	"startColumn": 64,
	"endLineNumber": 129,
	"endColumn": 68
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 129,
	"startColumn": 89,
	"endLineNumber": 129,
	"endColumn": 93
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportAssignmentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAssignmentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"Path\" is not assignable to declared type \"str\"\n  \"Path\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 135,
	"startColumn": 27,
	"endLineNumber": 135,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"_WorksheetOrChartsheetLike | None\" cannot be assigned to parameter \"worksheet\" of type \"_WorkbookSheet\" in function \"remove\"\n  Type \"_WorksheetOrChartsheetLike | None\" is not assignable to type \"_WorkbookSheet\"\n    Type \"None\" is not assignable to type \"_WorkbookSheet\"\n      \"None\" is not assignable to \"Worksheet\"\n      \"None\" is not assignable to \"WriteOnlyWorksheet\"\n      \"None\" is not assignable to \"ReadOnlyWorksheet\"\n      \"None\" is not assignable to \"Chartsheet\"",
	"source": "Pylance",
	"startLineNumber": 142,
	"startColumn": 19,
	"endLineNumber": 142,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 169,
	"startColumn": 48,
	"endLineNumber": 169,
	"endColumn": 52
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 169,
	"startColumn": 71,
	"endLineNumber": 169,
	"endColumn": 75
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"shape\" for class \"BarChart\"\n  Attribute \"shape\" is unknown",
	"source": "Pylance",
	"startLineNumber": 406,
	"startColumn": 15,
	"endLineNumber": 406,
	"endColumn": 20
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 598,
	"startColumn": 56,
	"endLineNumber": 598,
	"endColumn": 60
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 598,
	"startColumn": 79,
	"endLineNumber": 598,
	"endColumn": 83
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"List[int]\"\n  \"None\" is not assignable to \"List[int]\"",
	"source": "Pylance",
	"startLineNumber": 599,
	"startColumn": 60,
	"endLineNumber": 599,
	"endColumn": 64
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/excel_exporter.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 599,
	"startColumn": 85,
	"endLineNumber": 599,
	"endColumn": 89
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportIncompatibleMethodOverride",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Method \"beforeDrawPage\" overrides class \"PageTemplate\" in an incompatible manner\n  Parameter 2 name mismatch: base parameter is named \"canv\", override parameter is named \"canvas\"",
	"source": "Pylance",
	"startLineNumber": 45,
	"startColumn": 9,
	"endLineNumber": 45,
	"endColumn": 23,
	"relatedInformation": [
		{
			"startLineNumber": 154,
			"startColumn": 9,
			"endLineNumber": 154,
			"endColumn": 23,
			"message": "Overridden method",
			"resource": "/c:/Users/A/.vscode/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stubs/reportlab/reportlab/platypus/doctemplate.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"drawCentredText\" for class \"Canvas\"\n  Attribute \"drawCentredText\" is unknown",
	"source": "Pylance",
	"startLineNumber": 125,
	"startColumn": 16,
	"endLineNumber": 125,
	"endColumn": 31
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"drawCentredText\" for class \"Canvas\"\n  Attribute \"drawCentredText\" is unknown",
	"source": "Pylance",
	"startLineNumber": 132,
	"startColumn": 16,
	"endLineNumber": 132,
	"endColumn": 31
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 187,
	"startColumn": 72,
	"endLineNumber": 187,
	"endColumn": 76
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"Dict[Unknown, Unknown]\"\n  \"None\" is not assignable to \"Dict[Unknown, Unknown]\"",
	"source": "Pylance",
	"startLineNumber": 188,
	"startColumn": 53,
	"endLineNumber": 188,
	"endColumn": 57
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportAssignmentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAssignmentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"Path\" is not assignable to declared type \"str\"\n  \"Path\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 209,
	"startColumn": 31,
	"endLineNumber": 209,
	"endColumn": 52
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"_total_pages\" for class \"BaseDocTemplate\"\n  Attribute \"_total_pages\" is unknown",
	"source": "Pylance",
	"startLineNumber": 279,
	"startColumn": 13,
	"endLineNumber": 279,
	"endColumn": 25
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"_current_page\" for class \"BaseDocTemplate\"\n  Attribute \"_current_page\" is unknown",
	"source": "Pylance",
	"startLineNumber": 280,
	"startColumn": 13,
	"endLineNumber": 280,
	"endColumn": 26
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[date]\" cannot be assigned to parameter \"date_value\" of type \"date | datetime | str\" in function \"format_date_short\"\n  Type \"Column[date]\" is not assignable to type \"date | datetime | str\"\n    \"Column[date]\" is not assignable to \"date\"\n    \"Column[date]\" is not assignable to \"datetime\"\n    \"Column[date]\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 381,
	"startColumn": 45,
	"endLineNumber": 381,
	"endColumn": 65
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 473,
	"startColumn": 58,
	"endLineNumber": 473,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportAssignmentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAssignmentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"Path\" is not assignable to declared type \"str\"\n  \"Path\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 476,
	"startColumn": 32,
	"endLineNumber": 476,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 493,
	"startColumn": 61,
	"endLineNumber": 493,
	"endColumn": 65
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 509,
	"startColumn": 62,
	"endLineNumber": 509,
	"endColumn": 66
},{
	"resource": "/e:/kp/python/InvoiceApp/reports/pdf_generator.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 513,
	"startColumn": 84,
	"endLineNumber": 513,
	"endColumn": 88
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 121,
	"startColumn": 35,
	"endLineNumber": 121,
	"endColumn": 39
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"get\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 175,
	"startColumn": 39,
	"endLineNumber": 175,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"decode\" for class \"Awaitable[Any]\"\n  Attribute \"decode\" is unknown",
	"source": "Pylance",
	"startLineNumber": 177,
	"startColumn": 41,
	"endLineNumber": 177,
	"endColumn": 47
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"setex\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 191,
	"startColumn": 35,
	"endLineNumber": 191,
	"endColumn": 40
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"set\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 193,
	"startColumn": 35,
	"endLineNumber": 193,
	"endColumn": 38
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"delete\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 203,
	"startColumn": 43,
	"endLineNumber": 203,
	"endColumn": 49
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"flushdb\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 214,
	"startColumn": 31,
	"endLineNumber": 214,
	"endColumn": 38
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 283,
	"startColumn": 36,
	"endLineNumber": 283,
	"endColumn": 40
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Function with declared return type \"List[Dict[Unknown, Unknown]]\" must return value on all code paths\n  \"None\" is not assignable to \"List[Dict[Unknown, Unknown]]\"",
	"source": "Pylance",
	"startLineNumber": 352,
	"startColumn": 39,
	"endLineNumber": 352,
	"endColumn": 49
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Function with declared return type \"List[Dict[Unknown, Unknown]]\" must return value on all code paths\n  \"None\" is not assignable to \"List[Dict[Unknown, Unknown]]\"",
	"source": "Pylance",
	"startLineNumber": 358,
	"startColumn": 41,
	"endLineNumber": 358,
	"endColumn": 51
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Function with declared return type \"List[Dict[Unknown, Unknown]]\" must return value on all code paths\n  \"None\" is not assignable to \"List[Dict[Unknown, Unknown]]\"",
	"source": "Pylance",
	"startLineNumber": 363,
	"startColumn": 52,
	"endLineNumber": 363,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Function with declared return type \"Dict[str, Any]\" must return value on all code paths\n  \"None\" is not assignable to \"Dict[str, Any]\"",
	"source": "Pylance",
	"startLineNumber": 368,
	"startColumn": 43,
	"endLineNumber": 368,
	"endColumn": 57
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 372,
	"startColumn": 58,
	"endLineNumber": 372,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/services/cache_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 405,
	"startColumn": 58,
	"endLineNumber": 405,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"Image\" is unknown import symbol",
	"source": "Pylance",
	"startLineNumber": 29,
	"startColumn": 30,
	"endLineNumber": 29,
	"endColumn": 35
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 98,
	"startColumn": 70,
	"endLineNumber": 98,
	"endColumn": 74
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportAssignmentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAssignmentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"Path\" is not assignable to declared type \"str\"\n  \"Path\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 118,
	"startColumn": 31,
	"endLineNumber": 118,
	"endColumn": 52
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[date]\" cannot be assigned to parameter \"date_value\" of type \"date | datetime | str\" in function \"format_date_long\"\n  Type \"Column[date]\" is not assignable to type \"date | datetime | str\"\n    \"Column[date]\" is not assignable to \"date\"\n    \"Column[date]\" is not assignable to \"datetime\"\n    \"Column[date]\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 196,
	"startColumn": 60,
	"endLineNumber": 196,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[date]\" cannot be assigned to parameter \"date_value\" of type \"date | datetime | str\" in function \"format_date_short\"\n  Type \"Column[date]\" is not assignable to type \"date | datetime | str\"\n    \"Column[date]\" is not assignable to \"date\"\n    \"Column[date]\" is not assignable to \"datetime\"\n    \"Column[date]\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 267,
	"startColumn": 35,
	"endLineNumber": 267,
	"endColumn": 55
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int\" cannot be assigned to parameter \"amount\" of type \"Decimal | float | str | int\" in function \"format_currency_idr\"\n  Type \"Column[Decimal] | int\" is not assignable to type \"Decimal | float | str | int\"\n    Type \"Column[Decimal]\" is not assignable to type \"Decimal | float | str | int\"\n      \"Column[Decimal]\" is not assignable to \"Decimal\"\n      \"Column[Decimal]\" is not assignable to \"float\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 313,
	"startColumn": 51,
	"endLineNumber": 313,
	"endColumn": 67
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"amount\" of type \"Decimal | float | str | int\" in function \"format_currency_idr\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"Decimal | float | str | int\"\n    Type \"Column[Decimal]\" is not assignable to type \"Decimal | float | str | int\"\n      \"Column[Decimal]\" is not assignable to \"Decimal\"\n      \"Column[Decimal]\" is not assignable to \"float\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 314,
	"startColumn": 73,
	"endLineNumber": 314,
	"endColumn": 91
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"amount\" of type \"Decimal | float | str | int\" in function \"format_currency_idr\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"Decimal | float | str | int\"\n    Type \"Column[Decimal]\" is not assignable to type \"Decimal | float | str | int\"\n      \"Column[Decimal]\" is not assignable to \"Decimal\"\n      \"Column[Decimal]\" is not assignable to \"float\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 315,
	"startColumn": 48,
	"endLineNumber": 315,
	"endColumn": 68
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"amount\" of type \"Decimal | float | str | int\" in function \"format_currency_words\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"Decimal | float | str | int\"\n    Type \"Column[Decimal]\" is not assignable to type \"Decimal | float | str | int\"\n      \"Column[Decimal]\" is not assignable to \"Decimal\"\n      \"Column[Decimal]\" is not assignable to \"float\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 331,
	"startColumn": 60,
	"endLineNumber": 331,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 392,
	"startColumn": 81,
	"endLineNumber": 392,
	"endColumn": 85
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportAssignmentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAssignmentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"Path\" is not assignable to declared type \"str\"\n  \"Path\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 407,
	"startColumn": 27,
	"endLineNumber": 407,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"_WorksheetOrChartsheetLike | None\" cannot be assigned to parameter \"worksheet\" of type \"_WorkbookSheet\" in function \"remove\"\n  Type \"_WorksheetOrChartsheetLike | None\" is not assignable to type \"_WorkbookSheet\"\n    Type \"None\" is not assignable to type \"_WorkbookSheet\"\n      \"None\" is not assignable to \"Worksheet\"\n      \"None\" is not assignable to \"WriteOnlyWorksheet\"\n      \"None\" is not assignable to \"ReadOnlyWorksheet\"\n      \"None\" is not assignable to \"Chartsheet\"",
	"source": "Pylance",
	"startLineNumber": 413,
	"startColumn": 19,
	"endLineNumber": 413,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 454,
	"startColumn": 23,
	"endLineNumber": 454,
	"endColumn": 39
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"",
	"source": "Pylance",
	"startLineNumber": 454,
	"startColumn": 23,
	"endLineNumber": 454,
	"endColumn": 39
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 455,
	"startColumn": 23,
	"endLineNumber": 455,
	"endColumn": 41
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"",
	"source": "Pylance",
	"startLineNumber": 455,
	"startColumn": 23,
	"endLineNumber": 455,
	"endColumn": 41
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"\n  ...",
	"source": "Pylance",
	"startLineNumber": 456,
	"startColumn": 23,
	"endLineNumber": 456,
	"endColumn": 43
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[Decimal] | int | Unknown\" cannot be assigned to parameter \"x\" of type \"ConvertibleToFloat\" in function \"__new__\"\n  Type \"Column[Decimal] | int | Unknown\" is not assignable to type \"ConvertibleToFloat\"\n    Type \"Column[Decimal]\" is not assignable to type \"ConvertibleToFloat\"\n      \"Column[Decimal]\" is not assignable to \"str\"\n      \"Column[Decimal]\" is incompatible with protocol \"Buffer\"\n        \"__buffer__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsFloat\"\n        \"__float__\" is not present\n      \"Column[Decimal]\" is incompatible with protocol \"SupportsIndex\"",
	"source": "Pylance",
	"startLineNumber": 456,
	"startColumn": 23,
	"endLineNumber": 456,
	"endColumn": 43
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 556,
	"startColumn": 70,
	"endLineNumber": 556,
	"endColumn": 74
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 560,
	"startColumn": 80,
	"endLineNumber": 560,
	"endColumn": 84
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 594,
	"startColumn": 60,
	"endLineNumber": 594,
	"endColumn": 64
},{
	"resource": "/e:/kp/python/InvoiceApp/services/export_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 598,
	"startColumn": 70,
	"endLineNumber": 598,
	"endColumn": 74
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"safe_filename\" is unknown import symbol",
	"source": "Pylance",
	"startLineNumber": 33,
	"startColumn": 49,
	"endLineNumber": 33,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 40,
	"startColumn": 56,
	"endLineNumber": 40,
	"endColumn": 60
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 40,
	"startColumn": 75,
	"endLineNumber": 40,
	"endColumn": 79
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 57,
	"startColumn": 57,
	"endLineNumber": 57,
	"endColumn": 61
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 57,
	"startColumn": 76,
	"endLineNumber": 57,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 67,
	"startColumn": 59,
	"endLineNumber": 67,
	"endColumn": 63
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 67,
	"startColumn": 78,
	"endLineNumber": 67,
	"endColumn": 82
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 117,
	"startColumn": 65,
	"endLineNumber": 117,
	"endColumn": 69
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalSubscript",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalSubscript.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Object of type \"None\" is not subscriptable",
	"source": "Pylance",
	"startLineNumber": 132,
	"startColumn": 25,
	"endLineNumber": 132,
	"endColumn": 34
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"iter_rows\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 140,
	"startColumn": 53,
	"endLineNumber": 140,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"str | float | int | datetime\" cannot be assigned to parameter \"value\" of type \"int\" in function \"__setitem__\"\n  Type \"str | float | int | datetime\" is not assignable to type \"int\"\n    \"datetime\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 148,
	"startColumn": 29,
	"endLineNumber": 148,
	"endColumn": 49
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"None\" cannot be assigned to parameter \"value\" of type \"int\" in function \"__setitem__\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 150,
	"startColumn": 29,
	"endLineNumber": 150,
	"endColumn": 49
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 160,
	"startColumn": 72,
	"endLineNumber": 160,
	"endColumn": 76
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"entity_id\" of type \"int\" in function \"add_success\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 202,
	"startColumn": 40,
	"endLineNumber": 202,
	"endColumn": 50
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 222,
	"startColumn": 74,
	"endLineNumber": 222,
	"endColumn": 78
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Any | None\" cannot be assigned to parameter \"gender\" of type \"str\" in function \"_normalize_gender\"\n  Type \"Any | None\" is not assignable to type \"str\"\n    \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 238,
	"startColumn": 65,
	"endLineNumber": 238,
	"endColumn": 120
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"entity_id\" of type \"int\" in function \"add_success\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 263,
	"startColumn": 40,
	"endLineNumber": 263,
	"endColumn": 53
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 283,
	"startColumn": 79,
	"endLineNumber": 283,
	"endColumn": 83
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"entity_id\" of type \"int\" in function \"add_success\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 331,
	"startColumn": 40,
	"endLineNumber": 331,
	"endColumn": 51
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 351,
	"startColumn": 85,
	"endLineNumber": 351,
	"endColumn": 89
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"entity_id\" of type \"int\" in function \"add_success\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 443,
	"startColumn": 48,
	"endLineNumber": 443,
	"endColumn": 58
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"str | None\" cannot be assigned to parameter \"field\" of type \"str\" in function \"add_error\"\n  Type \"str | None\" is not assignable to type \"str\"\n    \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 446,
	"startColumn": 80,
	"endLineNumber": 446,
	"endColumn": 98
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportPossiblyUnboundVariable",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportPossiblyUnboundVariable.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"row_number\" is possibly unbound",
	"source": "Pylance",
	"startLineNumber": 449,
	"startColumn": 94,
	"endLineNumber": 449,
	"endColumn": 104
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportOperatorIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOperatorIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Operator \"+\" not supported for types \"Hashable\" and \"Literal[2]\"",
	"source": "Pylance",
	"startLineNumber": 534,
	"startColumn": 44,
	"endLineNumber": 534,
	"endColumn": 53
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Hashable\" cannot be assigned to parameter \"key\" of type \"str\" in function \"__setitem__\"\n  \"Hashable\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 537,
	"startColumn": 25,
	"endLineNumber": 537,
	"endColumn": 38
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Hashable\" cannot be assigned to parameter \"key\" of type \"str\" in function \"__setitem__\"\n  \"Hashable\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 539,
	"startColumn": 25,
	"endLineNumber": 539,
	"endColumn": 38
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"entity_id\" of type \"int\" in function \"add_success\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 599,
	"startColumn": 36,
	"endLineNumber": 599,
	"endColumn": 46
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Unknown | None\" cannot be assigned to parameter \"gender\" of type \"str\" in function \"_normalize_gender\"\n  Type \"Unknown | None\" is not assignable to type \"str\"\n    \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 624,
	"startColumn": 75,
	"endLineNumber": 624,
	"endColumn": 130
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"entity_id\" of type \"int\" in function \"add_success\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 646,
	"startColumn": 36,
	"endLineNumber": 646,
	"endColumn": 49
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"entity_id\" of type \"int\" in function \"add_success\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 700,
	"startColumn": 36,
	"endLineNumber": 700,
	"endColumn": 47
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 728,
	"startColumn": 76,
	"endLineNumber": 728,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 729,
	"startColumn": 39,
	"endLineNumber": 729,
	"endColumn": 43
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 758,
	"startColumn": 78,
	"endLineNumber": 758,
	"endColumn": 82
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Unknown | None\" cannot be assigned to parameter \"sheet_name\" of type \"str\" in function \"import_companies_excel\"\n  Type \"Unknown | None\" is not assignable to type \"str\"\n    \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 763,
	"startColumn": 73,
	"endLineNumber": 763,
	"endColumn": 83
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Unknown | None\" cannot be assigned to parameter \"sheet_name\" of type \"str\" in function \"import_tka_workers_excel\"\n  Type \"Unknown | None\" is not assignable to type \"str\"\n    \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 765,
	"startColumn": 75,
	"endLineNumber": 765,
	"endColumn": 85
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Unknown | None\" cannot be assigned to parameter \"sheet_name\" of type \"str\" in function \"import_job_descriptions_excel\"\n  Type \"Unknown | None\" is not assignable to type \"str\"\n    \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 767,
	"startColumn": 80,
	"endLineNumber": 767,
	"endColumn": 90
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Unknown | None\" cannot be assigned to parameter \"sheet_name\" of type \"str\" in function \"import_invoices_excel\"\n  Type \"Unknown | None\" is not assignable to type \"str\"\n    \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 771,
	"startColumn": 81,
	"endLineNumber": 771,
	"endColumn": 91
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 775,
	"startColumn": 76,
	"endLineNumber": 775,
	"endColumn": 80
},{
	"resource": "/e:/kp/python/InvoiceApp/services/import_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 794,
	"startColumn": 71,
	"endLineNumber": 794,
	"endColumn": 75
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"Session\"\n  \"None\" is not assignable to \"Session\"",
	"source": "Pylance",
	"startLineNumber": 34,
	"startColumn": 43,
	"endLineNumber": 34,
	"endColumn": 47
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 70,
	"startColumn": 24,
	"endLineNumber": 70,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 75,
	"startColumn": 24,
	"endLineNumber": 75,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 87,
	"startColumn": 24,
	"endLineNumber": 87,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"invoice_number\" for class \"Invoice\"\n  Expression of type \"str\" cannot be assigned to attribute \"invoice_number\" of class \"Invoice\"\n    \"str\" is not assignable to \"Column[str]\"",
	"source": "Pylance",
	"startLineNumber": 102,
	"startColumn": 25,
	"endLineNumber": 102,
	"endColumn": 39
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[date]\" cannot be assigned to parameter \"invoice_date\" of type \"date\" in function \"generate_invoice_number\"\n  \"Column[date]\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 102,
	"startColumn": 85,
	"endLineNumber": 102,
	"endColumn": 105
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[int]\" cannot be assigned to parameter \"invoice_id\" of type \"int\" in function \"_create_invoice_line\"\n  \"Column[int]\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 111,
	"startColumn": 50,
	"endLineNumber": 111,
	"endColumn": 60
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 132,
	"startColumn": 20,
	"endLineNumber": 132,
	"endColumn": 24
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"price\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 140,
	"startColumn": 84,
	"endLineNumber": 140,
	"endColumn": 89
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"List[Dict[str, Any]]\"\n  \"None\" is not assignable to \"List[Dict[str, Any]]\"",
	"source": "Pylance",
	"startLineNumber": 161,
	"startColumn": 58,
	"endLineNumber": 161,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 179,
	"startColumn": 24,
	"endLineNumber": 179,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 187,
	"startColumn": 24,
	"endLineNumber": 187,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 193,
	"startColumn": 24,
	"endLineNumber": 193,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 213,
	"startColumn": 28,
	"endLineNumber": 213,
	"endColumn": 32
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 243,
	"startColumn": 20,
	"endLineNumber": 243,
	"endColumn": 24
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 253,
	"startColumn": 24,
	"endLineNumber": 253,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 261,
	"startColumn": 24,
	"endLineNumber": 261,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 279,
	"startColumn": 20,
	"endLineNumber": 279,
	"endColumn": 24
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"Dict[str, Any]\"\n  \"None\" is not assignable to \"Dict[str, Any]\"",
	"source": "Pylance",
	"startLineNumber": 335,
	"startColumn": 52,
	"endLineNumber": 335,
	"endColumn": 56
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"date\"\n  \"None\" is not assignable to \"date\"",
	"source": "Pylance",
	"startLineNumber": 389,
	"startColumn": 47,
	"endLineNumber": 389,
	"endColumn": 51
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 397,
	"startColumn": 24,
	"endLineNumber": 397,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportReturnType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportReturnType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Type \"tuple[None, ValidationResult]\" is not assignable to return type \"Tuple[Invoice, ValidationResult]\"\n  \"None\" is not assignable to \"Invoice\"",
	"source": "Pylance",
	"startLineNumber": 433,
	"startColumn": 20,
	"endLineNumber": 433,
	"endColumn": 24
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"Decimal\"\n  \"None\" is not assignable to \"Decimal\"",
	"source": "Pylance",
	"startLineNumber": 454,
	"startColumn": 59,
	"endLineNumber": 454,
	"endColumn": 63
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "No overloads for \"__init__\" match the provided arguments",
	"source": "Pylance",
	"startLineNumber": 529,
	"startColumn": 30,
	"endLineNumber": 529,
	"endColumn": 49,
	"relatedInformation": [
		{
			"startLineNumber": 1116,
			"startColumn": 9,
			"endLineNumber": 1116,
			"endColumn": 17,
			"message": "Overload 8 is the closest match",
			"resource": "/c:/Users/A/.vscode/extensions/ms-python.vscode-pylance-2025.5.1/dist/typeshed-fallback/stdlib/builtins.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/services/invoice_service.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"List[Row[Tuple[str, int]]]\" cannot be assigned to parameter \"iterable\" of type \"Iterable[list[bytes]]\" in function \"__init__\"\n  \"List[Row[Tuple[str, int]]]\" is not assignable to \"Iterable[list[bytes]]\"\n    Type parameter \"_T_co@Iterable\" is covariant, but \"Row[Tuple[str, int]]\" is not a subtype of \"list[bytes]\"\n      \"Row[Tuple[str, int]]\" is not assignable to \"list[bytes]\"",
	"source": "Pylance",
	"startLineNumber": 529,
	"startColumn": 35,
	"endLineNumber": 529,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"setText\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 79,
	"startColumn": 19,
	"endLineNumber": 79,
	"endColumn": 26
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"setEnabled\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 84,
	"startColumn": 19,
	"endLineNumber": 84,
	"endColumn": 29
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"Company\"\n  \"None\" is not assignable to \"Company\"",
	"source": "Pylance",
	"startLineNumber": 171,
	"startColumn": 43,
	"endLineNumber": 171,
	"endColumn": 47
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"a0\" of type \"str | None\" in function \"setText\"\n  Type \"Column[str]\" is not assignable to type \"str | None\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 276,
	"startColumn": 45,
	"endLineNumber": 276,
	"endColumn": 70
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"a0\" of type \"str | None\" in function \"setText\"\n  Type \"Column[str]\" is not assignable to type \"str | None\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 277,
	"startColumn": 37,
	"endLineNumber": 277,
	"endColumn": 54
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"a0\" of type \"str | None\" in function \"setText\"\n  Type \"Column[str]\" is not assignable to type \"str | None\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 278,
	"startColumn": 38,
	"endLineNumber": 278,
	"endColumn": 56
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"text\" of type \"str | None\" in function \"setPlainText\"\n  Type \"Column[str]\" is not assignable to type \"str | None\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 279,
	"startColumn": 45,
	"endLineNumber": 279,
	"endColumn": 65
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[bool]\" cannot be assigned to parameter \"a0\" of type \"bool\" in function \"setChecked\"\n  \"Column[bool]\" is not assignable to \"bool\"",
	"source": "Pylance",
	"startLineNumber": 280,
	"startColumn": 45,
	"endLineNumber": 280,
	"endColumn": 67
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"TkaWorker\"\n  \"None\" is not assignable to \"TkaWorker\"",
	"source": "Pylance",
	"startLineNumber": 308,
	"startColumn": 48,
	"endLineNumber": 308,
	"endColumn": 52
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"a0\" of type \"str | None\" in function \"setText\"\n  Type \"Column[str]\" is not assignable to type \"str | None\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 431,
	"startColumn": 37,
	"endLineNumber": 431,
	"endColumn": 57
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"a0\" of type \"str | None\" in function \"setText\"\n  Type \"Column[str]\" is not assignable to type \"str | None\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 432,
	"startColumn": 41,
	"endLineNumber": 432,
	"endColumn": 65
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str]\" cannot be assigned to parameter \"text\" of type \"str | None\" in function \"setCurrentText\"\n  Type \"Column[str]\" is not assignable to type \"str | None\"\n    \"Column[str]\" is not assignable to \"str\"\n    \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 433,
	"startColumn": 46,
	"endLineNumber": 433,
	"endColumn": 75
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[str] | Literal['']\" cannot be assigned to parameter \"a0\" of type \"str | None\" in function \"setText\"\n  Type \"Column[str] | Literal['']\" is not assignable to type \"str | None\"\n    Type \"Column[str]\" is not assignable to type \"str | None\"\n      \"Column[str]\" is not assignable to \"str\"\n      \"Column[str]\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 434,
	"startColumn": 39,
	"endLineNumber": 434,
	"endColumn": 67
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Column[bool]\" cannot be assigned to parameter \"a0\" of type \"bool\" in function \"setChecked\"\n  \"Column[bool]\" is not assignable to \"bool\"",
	"source": "Pylance",
	"startLineNumber": 435,
	"startColumn": 45,
	"endLineNumber": 435,
	"endColumn": 70
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"toPython\" for class \"QDate\"\n  Attribute \"toPython\" is unknown",
	"source": "Pylance",
	"startLineNumber": 750,
	"startColumn": 54,
	"endLineNumber": 750,
	"endColumn": 62
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"Company\"\n  \"None\" is not assignable to \"Company\"",
	"source": "Pylance",
	"startLineNumber": 764,
	"startColumn": 44,
	"endLineNumber": 764,
	"endColumn": 48
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/dialogs.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"TkaWorker\"\n  \"None\" is not assignable to \"TkaWorker\"",
	"source": "Pylance",
	"startLineNumber": 771,
	"startColumn": 52,
	"endLineNumber": 771,
	"endColumn": 56
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportUndefinedVariable",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportUndefinedVariable.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"QComboBox\" is not defined",
	"source": "Pylance",
	"startLineNumber": 212,
	"startColumn": 30,
	"endLineNumber": 212,
	"endColumn": 39
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportUndefinedVariable",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportUndefinedVariable.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"QComboBox\" is not defined",
	"source": "Pylance",
	"startLineNumber": 217,
	"startColumn": 28,
	"endLineNumber": 217,
	"endColumn": 37
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addMenu\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 748,
	"startColumn": 29,
	"endLineNumber": 748,
	"endColumn": 36
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 753,
	"startColumn": 19,
	"endLineNumber": 753,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addSeparator\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 755,
	"startColumn": 19,
	"endLineNumber": 755,
	"endColumn": 31
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 759,
	"startColumn": 19,
	"endLineNumber": 759,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 763,
	"startColumn": 19,
	"endLineNumber": 763,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addSeparator\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 765,
	"startColumn": 19,
	"endLineNumber": 765,
	"endColumn": 31
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 770,
	"startColumn": 19,
	"endLineNumber": 770,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addMenu\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 773,
	"startColumn": 29,
	"endLineNumber": 773,
	"endColumn": 36
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 778,
	"startColumn": 19,
	"endLineNumber": 778,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addMenu\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 781,
	"startColumn": 29,
	"endLineNumber": 781,
	"endColumn": 36
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 785,
	"startColumn": 19,
	"endLineNumber": 785,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"setToolButtonStyle\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 790,
	"startColumn": 17,
	"endLineNumber": 790,
	"endColumn": 35
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 795,
	"startColumn": 17,
	"endLineNumber": 795,
	"endColumn": 26
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addSeparator\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 797,
	"startColumn": 17,
	"endLineNumber": 797,
	"endColumn": 29
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 802,
	"startColumn": 17,
	"endLineNumber": 802,
	"endColumn": 26
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addAction\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 807,
	"startColumn": 17,
	"endLineNumber": 807,
	"endColumn": 26
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"showMessage\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 812,
	"startColumn": 25,
	"endLineNumber": 812,
	"endColumn": 36
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"addPermanentWidget\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 817,
	"startColumn": 25,
	"endLineNumber": 817,
	"endColumn": 43
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"showMessage\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 835,
	"startColumn": 33,
	"endLineNumber": 835,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"showMessage\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 867,
	"startColumn": 29,
	"endLineNumber": 867,
	"endColumn": 40
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"refresh_data\" for class \"QWidget\"\n  Attribute \"refresh_data\" is unknown",
	"source": "Pylance",
	"startLineNumber": 892,
	"startColumn": 28,
	"endLineNumber": 892,
	"endColumn": 40
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"refresh_data\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 892,
	"startColumn": 28,
	"endLineNumber": 892,
	"endColumn": 40
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"refresh_invoices\" for class \"QWidget\"\n  Attribute \"refresh_invoices\" is unknown",
	"source": "Pylance",
	"startLineNumber": 894,
	"startColumn": 28,
	"endLineNumber": 894,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"refresh_invoices\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 894,
	"startColumn": 28,
	"endLineNumber": 894,
	"endColumn": 44
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"refresh_companies\" for class \"QWidget\"\n  Attribute \"refresh_companies\" is unknown",
	"source": "Pylance",
	"startLineNumber": 896,
	"startColumn": 28,
	"endLineNumber": 896,
	"endColumn": 45
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"refresh_companies\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 896,
	"startColumn": 28,
	"endLineNumber": 896,
	"endColumn": 45
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"refresh_workers\" for class \"QWidget\"\n  Attribute \"refresh_workers\" is unknown",
	"source": "Pylance",
	"startLineNumber": 898,
	"startColumn": 28,
	"endLineNumber": 898,
	"endColumn": 43
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"refresh_workers\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 898,
	"startColumn": 28,
	"endLineNumber": 898,
	"endColumn": 43
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"showMessage\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 900,
	"startColumn": 25,
	"endLineNumber": 900,
	"endColumn": 36
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/main_window.py",
	"owner": "python",
	"code": {
		"value": "reportIncompatibleMethodOverride",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Method \"closeEvent\" overrides class \"QWidget\" in an incompatible manner\n  Parameter 2 name mismatch: base parameter is named \"a0\", override parameter is named \"event\"",
	"source": "Pylance",
	"startLineNumber": 913,
	"startColumn": 9,
	"endLineNumber": 913,
	"endColumn": 19,
	"relatedInformation": [
		{
			"startLineNumber": 146,
			"startColumn": 9,
			"endLineNumber": 146,
			"endColumn": 19,
			"message": "Overridden method",
			"resource": "/e:/kp/python/InvoiceApp/venv/Lib/site-packages/PyQt6/QtWidgets.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"Literal['^[\\\\d,.]*\" cannot be assigned to parameter \"re\" of type \"QRegularExpression\" in function \"setRegularExpression\"]\" cannot be assigned to parameter \"re\" of type \"QRegularExpression\" in function \"setRegularExpression\"\n  \"Literal['^[\\\\d,.]*\" is not assignable to \"QRegularExpression\"]\" is not assignable to \"QRegularExpression\"",
	"source": "Pylance",
	"startLineNumber": 190,
	"startColumn": 45,
	"endLineNumber": 190,
	"endColumn": 57
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"completer\" for class \"AutoCompleteComboBox*\"\n  Type \"QCompleter\" is not assignable to type \"() -> (QCompleter | None)\"",
	"source": "Pylance",
	"startLineNumber": 247,
	"startColumn": 26,
	"endLineNumber": 247,
	"endColumn": 38
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportFunctionMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportFunctionMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"setCaseSensitivity\" for class \"function\"\n  Attribute \"setCaseSensitivity\" is unknown",
	"source": "Pylance",
	"startLineNumber": 248,
	"startColumn": 24,
	"endLineNumber": 248,
	"endColumn": 42
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportFunctionMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportFunctionMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"setFilterMode\" for class \"function\"\n  Attribute \"setFilterMode\" is unknown",
	"source": "Pylance",
	"startLineNumber": 249,
	"startColumn": 24,
	"endLineNumber": 249,
	"endColumn": 37
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Argument of type \"() -> (QCompleter | None)\" cannot be assigned to parameter \"c\" of type \"QCompleter | None\" in function \"setCompleter\"\n  Type \"() -> (QCompleter | None)\" is not assignable to type \"QCompleter | None\"\n    \"function\" is not assignable to \"QCompleter\"\n    \"function\" is not assignable to \"None\"",
	"source": "Pylance",
	"startLineNumber": 250,
	"startColumn": 27,
	"endLineNumber": 250,
	"endColumn": 41
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportFunctionMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportFunctionMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot access attribute \"setModel\" for class \"function\"\n  Attribute \"setModel\" is unknown",
	"source": "Pylance",
	"startLineNumber": 260,
	"startColumn": 24,
	"endLineNumber": 260,
	"endColumn": 32
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportIncompatibleMethodOverride",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Method \"textFromValue\" overrides class \"QDoubleSpinBox\" in an incompatible manner\n  Parameter 2 name mismatch: base parameter is named \"v\", override parameter is named \"value\"",
	"source": "Pylance",
	"startLineNumber": 272,
	"startColumn": 9,
	"endLineNumber": 272,
	"endColumn": 22,
	"relatedInformation": [
		{
			"startLineNumber": 6556,
			"startColumn": 9,
			"endLineNumber": 6556,
			"endColumn": 22,
			"message": "Overridden method",
			"resource": "/e:/kp/python/InvoiceApp/venv/Lib/site-packages/PyQt6/QtWidgets.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"setStretchLastSection\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 339,
	"startColumn": 16,
	"endLineNumber": 339,
	"endColumn": 37
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"setSectionResizeMode\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 340,
	"startColumn": 16,
	"endLineNumber": 340,
	"endColumn": 36
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"setVisible\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 343,
	"startColumn": 31,
	"endLineNumber": 343,
	"endColumn": 41
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportIncompatibleMethodOverride",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Method \"leaveEvent\" overrides class \"QWidget\" in an incompatible manner\n  Parameter 2 name mismatch: base parameter is named \"a0\", override parameter is named \"event\"",
	"source": "Pylance",
	"startLineNumber": 488,
	"startColumn": 9,
	"endLineNumber": 488,
	"endColumn": 19,
	"relatedInformation": [
		{
			"startLineNumber": 150,
			"startColumn": 9,
			"endLineNumber": 150,
			"endColumn": 19,
			"message": "Overridden method",
			"resource": "/e:/kp/python/InvoiceApp/venv/Lib/site-packages/PyQt6/QtWidgets.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportIncompatibleMethodOverride",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Method \"mousePressEvent\" overrides class \"QWidget\" in an incompatible manner\n  Parameter 2 name mismatch: base parameter is named \"a0\", override parameter is named \"event\"",
	"source": "Pylance",
	"startLineNumber": 493,
	"startColumn": 9,
	"endLineNumber": 493,
	"endColumn": 24,
	"relatedInformation": [
		{
			"startLineNumber": 160,
			"startColumn": 9,
			"endLineNumber": 160,
			"endColumn": 24,
			"message": "Overridden method",
			"resource": "/e:/kp/python/InvoiceApp/venv/Lib/site-packages/PyQt6/QtWidgets.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"unpolish\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 517,
	"startColumn": 22,
	"endLineNumber": 517,
	"endColumn": 30
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportOptionalMemberAccess",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOptionalMemberAccess.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "\"polish\" is not a known attribute of \"None\"",
	"source": "Pylance",
	"startLineNumber": 518,
	"startColumn": 22,
	"endLineNumber": 518,
	"endColumn": 28
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportAttributeAccessIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportAttributeAccessIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Cannot assign to attribute \"size\" for class \"LoadingSpinner*\"\n  Type \"int\" is not assignable to type \"() -> QSize\"",
	"source": "Pylance",
	"startLineNumber": 525,
	"startColumn": 21,
	"endLineNumber": 525,
	"endColumn": 25
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportIncompatibleMethodOverride",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Method \"paintEvent\" overrides class \"QWidget\" in an incompatible manner\n  Parameter 2 name mismatch: base parameter is named \"a0\", override parameter is named \"event\"",
	"source": "Pylance",
	"startLineNumber": 549,
	"startColumn": 9,
	"endLineNumber": 549,
	"endColumn": 19,
	"relatedInformation": [
		{
			"startLineNumber": 149,
			"startColumn": 9,
			"endLineNumber": 149,
			"endColumn": 19,
			"message": "Overridden method",
			"resource": "/e:/kp/python/InvoiceApp/venv/Lib/site-packages/PyQt6/QtWidgets.pyi"
		}
	]
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportOperatorIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOperatorIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Operator \"-\" not supported for type \"() -> QSize\"",
	"source": "Pylance",
	"startLineNumber": 572,
	"startColumn": 33,
	"endLineNumber": 572,
	"endColumn": 43
},{
	"resource": "/e:/kp/python/InvoiceApp/ui/widgets.py",
	"owner": "python",
	"code": {
		"value": "reportOperatorIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportOperatorIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Operator \"-\" not supported for type \"() -> QSize\"",
	"source": "Pylance",
	"startLineNumber": 572,
	"startColumn": 51,
	"endLineNumber": 572,
	"endColumn": 61
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/formatters.py",
	"owner": "python",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportGeneralTypeIssues.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expected class but received \"(iterable: Iterable[object], /) -> bool\"",
	"source": "Pylance",
	"startLineNumber": 414,
	"startColumn": 30,
	"endLineNumber": 414,
	"endColumn": 33
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/helpers.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"tuple[Unknown, ...]\"\n  \"None\" is not assignable to \"tuple[Unknown, ...]\"",
	"source": "Pylance",
	"startLineNumber": 436,
	"startColumn": 53,
	"endLineNumber": 436,
	"endColumn": 57
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/helpers.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"dict[Unknown, Unknown]\"\n  \"None\" is not assignable to \"dict[Unknown, Unknown]\"",
	"source": "Pylance",
	"startLineNumber": 436,
	"startColumn": 74,
	"endLineNumber": 436,
	"endColumn": 78
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/validators.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 53,
	"startColumn": 51,
	"endLineNumber": 53,
	"endColumn": 55
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/validators.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 53,
	"startColumn": 69,
	"endLineNumber": 53,
	"endColumn": 73
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/validators.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 66,
	"startColumn": 52,
	"endLineNumber": 66,
	"endColumn": 56
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/validators.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 66,
	"startColumn": 70,
	"endLineNumber": 66,
	"endColumn": 74
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/validators.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 75,
	"startColumn": 54,
	"endLineNumber": 75,
	"endColumn": 58
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/validators.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"str\"\n  \"None\" is not assignable to \"str\"",
	"source": "Pylance",
	"startLineNumber": 75,
	"startColumn": 72,
	"endLineNumber": 75,
	"endColumn": 76
},{
	"resource": "/e:/kp/python/InvoiceApp/utils/validators.py",
	"owner": "python",
	"code": {
		"value": "reportArgumentType",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expression of type \"None\" cannot be assigned to parameter of type \"int\"\n  \"None\" is not assignable to \"int\"",
	"source": "Pylance",
	"startLineNumber": 102,
	"startColumn": 79,
	"endLineNumber": 102,
	"endColumn": 83
}]